### PR TITLE
Move public options functions to separate file

### PR DESCRIPTION
**Pull request by nafur**
This PR moves the remaining special purpose functions out of the `Options` class. This set of functions is only used to implement API functions in the smt engine (getting and setting options by string), and by the main driver for parsing and printing usage information.

### CODE REVIEW

**Comment by nafur** [comment]
As per offline discussion, the current `options_public.*` will be gone soon, and the code moved in the PR will take its place. Marking wip until this has happened.

File: src/options/mkoptions.py [hunk 1]
```diff
@@ -234,6 +240,17 @@ def get_predicates(option):
     return ['opts.handler().{}("{}", option, value);'.format(x, optname)
             for x in option.predicates]
 
+def get_getall(module, option):
+    """Render snippet to add option to result of getAll()"""
+    if option.type == 'bool':
+        return 'res.push_back({{"{}", opts.{}.{} ? "true" : "false"}});'.format(option.long_name, module.id, option.name)
```
Comment (mpreiner) [inline src/options/mkoptions.py:246]: 80 chars

File: src/options/mkoptions.py [hunk 2]
```diff
@@ -234,6 +240,17 @@ def get_predicates(option):
...
+    elif is_numeric_cpp_type(option.type):
+        return 'res.push_back({{"{}", std::to_string(opts.{}.{})}});'.format(
+            option.long_name, module.id, option.name)
+    else:
+        return '{{ std::stringstream ss; ss << opts.{}.{}; res.push_back({{"{}", ss.str()}}); }}'.format(module.id,
```
Comment (mpreiner) [inline src/options/mkoptions.py:251]: Ditto.

File: src/options/options_public.h [hunk 3]
```diff
@@ -10,25 +10,77 @@
...
+ * Throws OptionException on failures.
+ *
+ * Preconditions: options and argv must be non-null.
+ */
+std::vector<std::string> parse(Options& opts,
+                               int argc,
+                               char* argv[],
+                               std::string& binaryName) CVC5_EXPORT;
+
+std::string get(const Options& opts, const std::string& key) CVC5_EXPORT;
```
Comment (mpreiner) [inline src/options/options_public.h:73]: Missing documentation.

File: src/options/options_public.h [hunk 4]
```diff
@@ -10,25 +10,77 @@
...
+
+void set(Options& opts,
```
Comment (mpreiner) [inline src/options/options_public.h:75]: Ditto.

File: src/options/mkoptions.py [hunk 5]
```diff
@@ -237,6 +243,19 @@ def get_predicates(option):
...
+def get_getall(module, option):
+    """Render snippet to add option to result of getAll()"""
+    if option.type == 'bool':
+        return 'res.push_back({{"{}", opts.{}.{} ? "true" : "false"}});'.format(
+            option.long_name, module.id, option.name)
+    elif is_numeric_cpp_type(option.type):
+        return 'res.push_back({{"{}", std::to_string(opts.{}.{})}});'.format(
+            option.long_name, module.id, option.name)
+    else:
+        return '{{ std::stringstream ss; ss << opts.{}.{}; res.push_back({{"{}", ss.str()}}); }}'.format(
```
Comment (mpreiner) [inline src/options/mkoptions.py:256]: 80 chars

File: src/options/mkoptions.py [hunk 6]
```diff
@@ -705,25 +724,25 @@ def add_getopt_long(long_name, argument_req, getopt_long):
             'required' if argument_req else 'no', value))
 
 
-def codegen_all_modules(modules, build_dir, dst_dir, tpl_options_h, tpl_options_cpp):
+def codegen_all_modules(modules, build_dir, dst_dir, tpl_options_h, tpl_options_cpp, tpl_options_public):
```
Comment (mpreiner) [inline src/options/mkoptions.py:727]: Ditto

**Review by mpreiner — APPROVED** [review]
Make sure to format the Python files to 80 chars.

**Review by mpreiner — APPROVED** [review]
LGTM, thanks for the changes!

**Pull request merged**